nebula-logger/main/log-management/classes/LogEntryEventHandler.cls

```diff
@@ -209,4 +225,86 @@ public without sharing class LogEntryEventHandler {
+    private static Log__c getRecentLogWithDetails() {
+        // Query for recent logs created only today - the status API should be called...
+        // ...at least once per day to make sure that status details are still accurate.
+        // This query should make a callout approximately every 4 hours.
+        Datetime fourHoursAgo = System.now().addMinutes(-4 * 60);
+
+        List<Log__c> logs = [
+            SELECT Id, ApiReleaseNumber__c, ApiReleaseVersion__c
+            FROM Log__c
+            WHERE CreatedDate >= :fourHoursAgo AND DAY_ONLY(CreatedDate) = TODAY AND ApiReleaseNumber__c != NULL
+            ORDER BY CreatedDate DESC
+            LIMIT 1
+        ];
+
+        if (logs.isEmpty()) {
+            return null;
+        } else {
+            return logs.get(0);
+        }
+    }
+
+    private static Integer getCountOfOpenJobs() {
+        return [
+            SELECT COUNT()
+            FROM AsyncApexJob
+            WHERE
+                ApexClass.Name = :LogEntryEventHandler.class.getName()
+                AND MethodName = 'setStatusApiDetails'
+                AND Status IN ('Holding', 'Queued', 'Preparing', 'Processing')
+        ];
+    }
+
+    @future(callout=true)
+    private static void setStatusApiDetails() {
+        System.debug('Running setStatusApiDetails()');
+
+        Organization organization = [SELECT InstanceName FROM Organization];
+        String statusApiEndpoint = 'https://api.status.salesforce.com/v1/instances/' + organization.InstanceName + '/status';
+
+        HttpRequest request = new HttpRequest();
+        request.setEndpoint(statusApiEndpoint);
+        request.setMethod('GET');
+
+        HttpResponse response = new Http().send(request);
+
+        if (response.getStatusCode() >= 400) {
+            String errorMessage =
+                'Callout failed for ' +
+                statusApiEndpoint +
+                '\nReceived request status code ' +
+                response.getStatusCode() +
+                ', status message: ' +
+                response.getStatus();
+            throw new StatusApiResponseException(errorMessage);
+        }
+
+        StatusApiResponse statusApiResponse = (StatusApiResponse) JSON.deserialize(response.getBody(), StatusApiResponse.class);
+        System.debug('statusApiResponse==' + statusApiResponse);
+
+        List<Log__c> logsToUpdate = new List<Log__c>();
+        for (Log__c log : [SELECT Id FROM Log__c WHERE DAY_ONLY(CreatedDate) = TODAY AND ApiReleaseNumber__c = NULL ORDER BY CreatedDate LIMIT :Limits.getLimitDmlRows()]) {
+            log.ApiReleaseNumber__c = statusApiResponse.releaseNumber;
+            log.ApiReleaseVersion__c = statusApiResponse.releaseVersion;
+
+            logsToUpdate.add(log);
+        }
+        System.debug('logsToUpdate==' + logsToUpdate);
+        update logsToUpdate;
+    }
+
+    // Private class for handling the response from api.status.salesforce.com
+    @TestVisible
+    private class StatusApiResponse {
```

Comment:
nice, no need to store _too_ much of that crazy JSON

Reply:
Yeah, and we can always add new properties to the class if we want to grab anything else from the API.